code/modules/events/disease_outbreak.dm

```diff
@@ -59,28 +59,29 @@
 			continue
 		disease_candidates += candidate
 
-/datum/event_admin_setup/disease_outbreak
-	///Admin selected disease, to be passed down to the round_event
-	var/virus_type
+///Handles checking and alerting admins about the number of valid candidates
+/datum/event_admin_setup/minimum_candidate_requirement/disease_outbreak
+	output_text = "There are no candidates eligible to recieve a disease!"
 
-/// Checks for candidates. Returns false if there isn't enough
-/datum/event_admin_setup/disease_outbreak/proc/candidate_check()
+/datum/event_admin_setup/minimum_candidate_requirement/disease_outbreak/count_candidates()
 	var/datum/round_event_control/disease_outbreak/disease_control = event_control
 	disease_control.generate_candidates() //can_spawn_event() is bypassed by admin_setup, so this makes sure that the candidates are still generated
 	return length(disease_control.disease_candidates)
 
-/datum/event_admin_setup/disease_outbreak/prompt_admins()
-	var/candidate_count = candidate_check()
-	if(!candidate_count)
-		tgui_alert(usr, "There are no candidates eligible to recieve a disease!", "Error")
-		return ADMIN_CANCEL_EVENT
-	tgui_alert(usr, "[candidate_count] candidates found!", "Disease Outbreak")
 
-	if(tgui_alert(usr, "Select a specific disease?", "Sickening behavior", list("Yes", "No")) == "Yes")
-		virus_type = tgui_input_list(usr, "Warning: Some of these are EXTREMELY dangerous.","Bacteria Hysteria", subtypesof(/datum/disease))
+///Handles actually selecting whicch disease will spawn.
+/datum/event_admin_setup/listed_options/disease_outbreak
+	input_text = "Select a specific disease? Warning: Some are EXTREMELY dangerous."
+	normal_run_option = "Random Classic Disease (Safe)"
+	special_run_option = "Entirely Random Disease (Dangerous)"
 
-/datum/event_admin_setup/disease_outbreak/apply_to_event(datum/round_event/disease_outbreak/event)
-	event.virus_type = virus_type
+/datum/event_admin_setup/listed_options/disease_outbreak/get_list()
+	return subtypesof(/datum/disease)
+
+/datum/event_admin_setup/listed_options/disease_outbreak/apply_to_event(datum/round_event/disease_outbreak/event)
+	if(chosen == special_run_option)
+		chosen = pick(get_list())
+	event.virus_type = chosen
```

Comment:
I feel like this is really pushing what chosen should be to be honest... feels silly that we start out with a string equivalency check to a user-facing message, then mutate it into a datum typepath just so you can apply it to the event

just create a new proc-local variable `var/datum/disease/virus` because i'm not the biggest fan on doing so many mutations to one variable that we might re-use later in a potentially different context, could be a bit more type-safe and all that

i'm okay with all of the other schenanigans that we might do with chosen in order to achieve a more abstract state, but I just really don't want it to be mutated (in this manner anyways) in case someone comes along one day and tries to extend something to end up getting smartphowned